go-controller/pkg/ovn/gress_policy.go

```diff
@@ -19,6 +20,8 @@ import (
 )
 
 type gressPolicy struct {
+	// used only to synchronize on deletion
+	sync.RWMutex
```

Comment:
let's bring back the debate on RWMutex perf ;) @aojea @dave-tucker 

https://github.com/golang/go/issues/17973

Reply:
Given that these are all controlling access to another standard lock, we can use a standard `Mutex` with no real loss of performance.

Reply:
That is to say, the underlying AddressSet also has a mutex.

Reply:
The scary thing here is the amount of nested locking ...

Reply:
You're not wrong.

I can make this race go away by doing copies instead - the AddressSet is already thread safe.

Reply:
there. No more locks.